From b9918bb6c0b96583a1f635bf5f216dce15b20515 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:39:33 -0400
Subject: [virt] kvm: Rename variable smep to cr4_smep

Message-id: <1406040016-3289-179-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86257
O-Subject: [RHEL7 PATCH v2 178/221] KVM: Rename variable smep to cr4_smep
Bugzilla: 1084616 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Feng Wu <feng.wu@intel.com>

Rename variable smep to cr4_smep, which can better reflect the
meaning of the variable.

Signed-off-by: Feng Wu <feng.wu@intel.com>
Signed-off-by: Marcelo Tosatti <mtosatti@redhat.com>
(cherry picked from commit 66386ade2aa7f5fcf0dda4171df7b4d4c274d2d6)
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kvm/mmu.c b/arch/x86/kvm/mmu.c
index 51034ea..d3aa064 100644
--- a/arch/x86/kvm/mmu.c
+++ b/arch/x86/kvm/mmu.c
@@ -3606,9 +3606,9 @@ void update_permission_bitmask(struct kvm_vcpu *vcpu,
 {
  unsigned bit, byte, pfec;
  u8 map;
- bool fault, x, w, u, wf, uf, ff, smapf, cr4_smap, smep, smap = 0;
+ bool fault, x, w, u, wf, uf, ff, smapf, cr4_smap, cr4_smep, smap = 0;
 
- smep = kvm_read_cr4_bits(vcpu, X86_CR4_SMEP);
+ cr4_smep = kvm_read_cr4_bits(vcpu, X86_CR4_SMEP);
  cr4_smap = kvm_read_cr4_bits(vcpu, X86_CR4_SMAP);
  for (byte = 0; byte < ARRAY_SIZE(mmu->permissions); ++byte) {
   pfec = byte << 1;
@@ -3633,7 +3633,7 @@ void update_permission_bitmask(struct kvm_vcpu *vcpu,
     /* Allow supervisor writes if !cr0.wp */
     w |= !is_write_protection(vcpu) && !uf;
     /* Disallow supervisor fetches of user code if cr4.smep */
-    x &= !(smep && u && !uf);
+    x &= !(cr4_smep && u && !uf);
 
     /*
      * SMAP:kernel-mode data accesses from user-mode
-- 
1.7.1